### ISSUE TITLE
Handling of text in Latin1 (ISO-8859-1) encoding

### ISSUE THREAD

**Issue by xspager** [SerenityOS/serenity#4233]
The presence of a file with Latin1 (ISO-8859-1) encoding or opening a website that send text in Latin1 break the browser and the text editor. I'm assuming any block of text that contains non ASCII characters and it's not encoded in UTF-8 will break stuff.

This assertion fails but it's perfectly sensible if the non ASCII text is in UTF-8:
```
USERSPACE(35) ASSERTION FAILED: m_ptr[offset] >> 6 == 2
./AK/Utf8View.cpp:208
```
I suppose everything is assuming a perfect world where everything is UTF-8.

**Comment by bcoles** [comment]
> I'm assuming any block of text that contains non ASCII characters and it's not encoded in UTF-8 will break stuff.

Probably

> I suppose everything is assuming a perfect world where everything is UTF-8.

Yeah there's a bunch of UTF8 related issues.

#1600
#3549

**Comment by alimpfard** [comment]
I was under the impression that the browser respected the reported encoding?
if that is true, then...are you setting its encoding to Latin-1?

**Comment by xspager** [comment]
I'll check now that can examine the response, but if I'm not mistaken
neither the encoding is sent in the HTTP header nor in the HTML meta tag

**Comment by awesomekling** [comment]
Indeed, we need to gracefully handle servers that don't provide the encoding, or lie about it, or just send random bytes.

**Comment by xspager** [comment]
I checked and yes, the server sends no header indicating the encoding.

The problem is not limited to Browser, if you just try to open a file that is not in ASCII or UTF-8 with the TextEditor it will crash.

**Comment by Lubrsi** [comment]
I think one of the main issues is that the browser and text editor don't try to work out the encoding by looking at the bytes. There's a section on it in the HTML parsing spec here: https://html.spec.whatwg.org/multipage/parsing.html#the-input-byte-stream
